src/app/modules/angular-slickgrid/extensions/draggableGroupingExtension.ts

```diff
@@ -65,6 +66,13 @@ export class DraggableGroupingExtension implements Extension {
             this.sharedService.gridOptions.draggableGrouping.onGroupChanged(e, args);
           }
         });
+
+        // if a column was added dynamically, we need to re-initialize the draggable grouping after the render
+        this._eventHandler.subscribe(grid.onRendered, () => {
+          if (this._addon && this.sharedService.dataView.getGroups && this.sharedService.dataView.getGroups().length === 0) {
```

Comment:
This was a bit of a hack on my part, and had the following problems:
1) It won't work if someone tries to programmatically set some groupings before the gird has rendered or keep some existing groupings in the event new columns are added (`sharedService.dataView.getGroups().length > 0`)
2) This code will be invoked every time the grid is rendered, when ideally we would trigger this only when the columns have changed

I have looked through the exposed API a lot but I haven't looked at the internals so unfortunately I cannot suggest a better solution at this time (maybe you know what events we can trigger from). I would perhaps say it is better to force people to implement the workaround rather than doing this potentially destructive re-init without the developers control.

Reply:
Maybe add a flag `isGridFirstRender` that will be set after the `onRendered` and we basically won't run this piece of code if that is the first render. Can you try it on your side first?

Reply:
So in my code, that would transform into something like this
```ts
isGridFirstRender = false;
// ...

this._eventHandler.subscribe(grid.onRendered, () => {
    if (!this.isGridFirstRender && this._addon && this.sharedService.dataView.getGroups && this.sharedService.dataView.getGroups().length === 0) {
    this._addon.init(grid);
    this.isGridFirstRender = true;
  }
});
```

Reply:
Unfortunately my example is in a secure environment where I cannot checkout and depend on this branch. I could potentially try it at home when I next get a chance. (Or if you have a suggestion for how to patch the node_module I can do that?)

Reply:
I meant for you to try in your local, not necessarily my code. Just add yourself that flag and see if it covers both of your point 1 & 2 that you wrote earlier.

Reply:
Ah, regarding point 1, in my example I have had to call 
```
if (this.draggableGroupingPlugin) {
    this.draggableGroupingPlugin.clearDroppedGroups();
}
```
whenever I change the column definitions, otherwise we fail the `.dataView.getGroups().length > 0` check and we never re-initialise the addOn. 
In a proper solution I would hope that we do not have to check `dataView.getGroups().length`, but if we have to re-initialise the component we should re-apply the grouping prior to the init method.

Reply:
I actually inversed the flag, I update the code I wrote earlier. Can you try that on your local with a group set?

Reply:
For what it is worth, I have also checked if the boolean flag solves point 2, and unfortunately it does not as the grid render callback is called multiple times, some of which are before the grid state has been updated to include the new columns. If you limit it to one invocation it doesn't work.

Reply:
From your last point, I'm not sure what you're asking me to change, can you provide possible code change?

This is what I have so far 
```ts
// if a column was added dynamically, we need to re-initialize the draggable grouping after the render
// but we want to run this only after the column definitions were changed and not on a regular page load
// so for that reason we'll only run it after the grid was rendered multiple times (that is when column definitions were changed dynamically)
this._eventHandler.subscribe(grid.onRendered, () => {
  if (!this._isGridFirstRender && this._addon && this.sharedService.dataView.getGroups && this.sharedService.dataView.getGroups().length === 0) {
    this._addon.init(grid);
  }
  this._isGridFirstRender = false;
});
```

Reply:
in that case, I only have 1 other code change that I could think of, when you as a user change the column definitions afterward, I have a flag `isGridInitialized` internally [here](https://github.com/ghiscoding/Angular-Slickgrid/blob/master/src/app/modules/angular-slickgrid/components/angular-slickgrid.component.ts#L173) in my lib which is on the `columnDefinitions` setter. I could possible add it into my shared service and then use it within the extension and the `onRendered`. 

```ts
this._eventHandler.subscribe(grid.onRendered, () => {
  if (!this.sharedService.isGridInitialized && this._addon && this.sharedService.dataView.getGroups && this.sharedService.dataView.getGroups().length === 0) {
    this._addon.init(grid);
  }
  this._isGridFirstRender = false;
});
```

Reply:
Thanks for your help. What I'll need to do is re-create my test case at home when I next have time to and then I can offer some concrete suggestions and try the code snippet you have posted above too. Feel free to close the issue with a wiki update for now and I'll raise a PR if I find a good solution?